dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultAnalyticsService.java

```diff
@@ -1401,6 +1486,26 @@ private void handleEmptyDimensionItemPermutations( List<List<DimensionItem>> dim
         }
     }
 
+    /**
+     * Transforms a List of {@see DimensionItemWithValue} into a Map where:
+     * 
+     * key -> DimensionalItemObject value -> the value of the DimensionalItemObject
+     * 
+     */
+    private Map<DimensionalItemObject, Double> convertToDimItemValueMap(
```

Comment:
Could we move this method to a utility class, add Javadoc and add unit test? E.g.AnalyticsUtils.

Reply:
Fix + test